src/background.js

```diff
@@ -1,3 +1,31 @@
+
+/*
+   Originally from Schalk Neethling at https://github.com/schalkneethling/dnt-helper
+ * Respects user choice & honor DoNotTrack
+ * Returns true or false based on whether doNotTack is enabled.
+ * @returns {boolean} true if enabled else false
+ */
+
+
+// browser.runtime.getManifest().version;
+
+function _dntEnabled(dnt, userAgent) {
```

Comment:
genuine question: does all this code work from within a web extension? My default approach to this would be to use Firefox apis/prefs to determine DNT status. But if this works, great!

Reply:
great question - it works when Tracking Protection is set to "always" and the "send DoNotTrack" signal is set to always. It does not however disable the add-on when a user is in a Private Window unless both of those preferences are explicitly set so if there is a way to identify a Private Window, or a better way to do this please share!

Reply:
Can we leave this doNotTrack stuff out? I'm not sure what the context/discussion was that led to this addition, but it doesn't seem right to me. do not track is a web feature that isn't really relevant to Firefox features like this. It will also completely exclude all users with tracking protection enabled. There is another pref very close to the do not track pref in about:preferences, which lets users opt out of all shield studies. Let me know what you think.

There is definitely a way to identify a private window. We should do that from FirefoxMonitor.jsm, in the warnIfNeeded function. See https://searchfox.org/mozilla-central/source/toolkit/modules/PrivateBrowsingUtils.jsm#42.